devbuild.sh

```diff
@@ -1,76 +1,235 @@
+  printf "      (.e.g. cheyenne | hera | jet | orion | wcoss)\n"
+  printf "  --compiler=COMPILER\n"
+  printf "      compiler to use; default depends on platform\n"
+  printf "      (.e.g. intel | gnu | cray | gccgfortran)\n"
+  printf "  --app=APPLICATION\n"
+  printf "      weather model application to build\n"
+  printf "      (.e.g. ATM | ATMW | S2S | S2SW)\n"
+  printf "  --ccpp=\"CCPP_SUITE1,CCPP_SUITE2...\"\n"
+  printf "      CCCP suites to include in build; delimited with ','\n"
+  printf "  --components=\"COMPONENT1,COMPONENT2...\"\n"
+  printf "      components to include in build; delimited with ','\n"
+  printf "  --continue\n"
+  printf "      continue with existing build\n"
+  printf "  --clean\n"
+  printf "      removes existing build; overrides --continue\n"
+  printf "  --build-dir=BUILD_DIR\n"
+  printf "      build directory\n"
+  printf "  --install-dir=INSTALL_DIR\n"
+  printf "      installation prefix\n"
+  printf "  --build-type=BUILD_TYPE\n"
+  printf "      build type; defaults to RELEASE\n"
+  printf "      (.e.g. DEBUG | RELEASE | RELWITHDEBINFO)\n"
+  printf "  --build-jobs=BUILD_JOBS\n"
+  printf "      number of build jobs; defaults to 4\n"
+  printf "  -v, --verbose\n"
+  printf "      build with verbose output\n"
+  printf "\n"
+  printf "NOTE: This script is for internal developer use only;\n"
+  printf "See User's Guide for detailed build instructions\n"
 }
 
-PLATFORM="${1:-NONE}"
-COMPILER="${2:-intel}"
+# print settings
+settings () {
+  printf "Settings:\n"
+  printf "\n"
+  printf "  SRC_DIR=${SRC_DIR}\n"
+  printf "  BUILD_DIR=${BUILD_DIR}\n"
+  printf "  INSTALL_DIR=${INSTALL_DIR}\n"
+  printf "  PLATFORM=${PLATFORM}\n"
+  printf "  COMPILER=${COMPILER}\n"
+  if [ ! -z "${APPLICATION}" ]; then printf "  APP=${APPLICATION}\n"; fi
+  if [ ! -z "${CCPP}" ]; then printf "  CCPP=${CCPP}\n"; fi
+  if [ ! -z "${COMPONENTS}" ]; then printf "  COMPONENTS=${COMPONENTS}\n"; fi
+  printf "  CLEAN=${CLEAN}\n"
+  printf "  CONTINUE=${CONTINUE}\n"
+  printf "  BUILD_TYPE=${BUILD_TYPE}\n"
+  printf "  BUILD_JOBS=${BUILD_JOBS}\n"
+  printf "  VERBOSE=${VERBOSE}\n"
+  printf "\n"
+}
 
+# find system name
+find_system () {
+    local sysname=`hostname`
+    sysname="${sysname//[[:digit:]]/}"
+    if [ "${sysname}" = "fe" ]; then
+      sysname="jet"
+    fi
+    echo "$sysname"
+}
 
-if [ $# -lt 1 ]; then 
-  echo "ERROR: not enough arguments"
-  usage
-  exit 1
+# default settings
+LCL_PID=$$
+SRC_DIR=$(cd "$(dirname "$(readlink -f -n "${BASH_SOURCE[0]}" )" )" && pwd -P)
+BUILD_DIR=${SRC_DIR}/build
+INSTALL_DIR=${SRC_DIR}
+PLATFORM=""
+COMPILER=""
+APPLICATION=""
+CCPP=""
+COMPONENTS=""
+BUILD_TYPE="RELEASE"
+BUILD_JOBS=4
+CLEAN=false
+CONTINUE=false
+VERBOSE=false
+
+# process arguments
+while :; do
+  case $1 in
+    --help|-h) usage; exit 0 ;;
+    --platform=?*) PLATFORM=${1#*=} ;;
+    --platform) printf "ERROR: $1 requires an argument.\n"; usage; exit 1 ;;
```

Comment:
Suggestion (that you don't have to take):  To make this list shorter (and have less repeated code), you can combine the cases without and with equal signs:
`--platform|--platform=) printf ...
`
You can also make a function for 
`printf ""ERROR: $1 requires an argument.\n"; usage; exit 1
`
and call that for the appropriate cases.

Reply:
I've combined and cleaned up error messages.